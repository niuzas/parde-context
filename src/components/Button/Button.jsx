import React from 'react';
import PropTypes from 'prop-types';
import * as S from './Button.style';

const Button = ({ color, type, handleClick, children }) => (
  <S.Button className={`button is-${color}`} type={type} onClick={handleClick}>
    {children}
  </S.Button>
);

const buttonStyles = ['primary', 'secondary'];

Button.propTypes = {
  color: PropTypes.oneOf(buttonStyles),
  type: PropTypes.string,
  handleClick: PropTypes.func.isRequired,
  children: PropTypes.string.isRequired
};

Button.defaultProps = {
  color: 'primary',
  type: 'button'
};

export default Button;
