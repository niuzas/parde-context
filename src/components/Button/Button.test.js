import React from 'react';
import renderer from 'react-test-renderer';

import Button from './Button';

it('Button renders correctly', () => {
  const tree = renderer.create(<Button handleClick={() => null}>Button</Button>).toJSON();
  expect(tree).toMatchSnapshot();
});
