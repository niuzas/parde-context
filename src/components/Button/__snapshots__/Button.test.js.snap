// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  className="sc-bdfBwQ dWDbCJ button is-primary"
  onClick={[Function]}
  type="button"
>
  Button
</button>
`;
